tap_core/tests/manager_test.rs

```diff
@@ -453,3 +516,46 @@ async fn manager_create_multiple_rav_requests_all_valid_receipts_consecutive_tim
         .await
         .is_ok());
 }
+
+#[rstest]
+#[tokio::test]
+async fn manager_create_rav_and_ignore_invalid_receipts(
+    keys: (LocalWallet, Address),
+    allocation_ids: Vec<Address>,
+    domain_separator: Eip712Domain,
+    context: ContextFixture,
+) {
+    let ContextFixture {
+        context,
+        checks,
+        escrow_storage,
+        ..
+    } = context;
+
+    let manager = Manager::new(domain_separator.clone(), context.clone(), checks);
+
+    escrow_storage.write().unwrap().insert(keys.1, 999999);
+
+    let mut stored_signed_receipts = Vec::new();
+    //Forcing all receipts but one to be invalid by making all the same
+    for _ in 0..10 {
+        let value = 20u128;
+        let mut receipt = Receipt::new(allocation_ids[0], value).unwrap();
+        receipt.nonce = 1;
+        receipt.timestamp_ns = 1;
+        let signed_receipt = EIP712SignedMessage::new(&domain_separator, receipt, &keys.0).unwrap();
+        stored_signed_receipts.push(signed_receipt.clone());
+        let _ = manager.verify_and_store_receipt(signed_receipt).await;
+    }
+
+    let rav_request_result = manager.create_rav_request(0, None).await;
+    assert!(rav_request_result.is_ok());
+
+    let rav_request = rav_request_result.unwrap();
+
+    assert!(rav_request.valid_receipts.len() < stored_signed_receipts.len());
```

Comment:
We can be more precise here
```suggestion
    assert_eq!(rav_request.valid_receipts.len(), 1);
```